test/plugins/BufferCreator.cpp

```diff
@@ -0,0 +1,185 @@
+  register_command("start", &BufferCreator::do_start);
+  register_command("stop", &BufferCreator::do_stop);
+  register_command("scrap", &BufferCreator::do_scrap);
+}
+
+void
+BufferCreator::init(const nlohmann::json& init_data)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering init() method";
+
+  m_input_queue_tps.reset(new appfwk::DAQSource<trigger::TPSet>(appfwk::queue_inst(init_data, "tpset_source")));
+  m_input_queue_dr.reset(new appfwk::DAQSource<std::vector<dataformats::timestamp_t>>(appfwk::queue_inst(init_data, "data_request_source")));
+  m_output_queue_frag.reset(new appfwk::DAQSink<dataformats::Fragment>(appfwk::queue_inst(init_data, "fragment_sink")));
+
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting init() method";
+}
+
+void
+BufferCreator::get_info(opmonlib::InfoCollector& /*ci*/, int /*level*/)
+{}
+
+void
+BufferCreator::do_configure(const nlohmann::json& obj)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_configure() method";
+
+  auto params = obj.get<buffercreator::Conf>();
+
+  m_sleep_time = params.sleep_time;
+  m_buffer_size = params.buffer_size;
+
+  m_buffer->SetBufferSize(m_buffer_size);
+
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_configure() method";
+}
+
+void
+BufferCreator::do_start(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_start() method";
+  m_thread.start_working_thread("buffer-man");
+  TLOG() << get_name() << " successfully started";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_start() method";
+}
+
+void
+BufferCreator::do_stop(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_stop() method";
+  m_thread.stop_working_thread();
+  TLOG() << get_name() << " successfully stopped";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_stop() method";
+}
+
+void
+BufferCreator::do_scrap(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_scrap() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_scrap() method";
+}
+
+dataformats::Fragment 
+BufferCreator::convert_to_fragment(std::vector<trigger::TPSet> tps)
+{
+  dataformats::Fragment frag(nullptr, tps.size());
+
+  return frag;
+}
+ 
+void
+BufferCreator::do_work(std::atomic<bool>& running_flag)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_work() method";
+  size_t addedCount = 0;
+  size_t requestedCount = 0;
+  size_t sentCount = 0;
+
+  auto start_time = std::chrono::steady_clock::now();
+  auto period = std::chrono::nanoseconds(m_sleep_time);
+  auto next_time_step = start_time + period;
+
+  while (running_flag.load()) {
+
+    TLOG_DEBUG(TLVL_GENERATION) << get_name() << ": Start of sleep between input/output";
+    std::this_thread::sleep_until(next_time_step);
```

Comment:
I don't think you want a sleep here. We want to handle inputs as fast as they arrive, without spinning the CPU at 100% when nothing is arriving. But the pop timeout on the queues does this for us: if no input arrives, pop() will sleep for up to `timeout`, so the CPU isn't spun

Reply:
True! Sleep was carried out from the other module I use as example to write this one. Removed now.